From e457053784d4fe6374ae377fc0861f316ebeacc5 Mon Sep 17 00:00:00 2001
From: Reto Schneider <reto.schneider@husqvarnagroup.com>
Date: Tue, 8 Feb 2022 16:19:53 +0100
Subject: [PATCH] [sg noup] fw_setenv: Support extra long lines

Our certificates are bigger than 1024 bytes (seen: 1722 bytes). Bumping the
line length to a value which certainly is enough.
---
 src/uboot_env.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/uboot_env.c b/src/uboot_env.c
index c9a900f..823a022 100644
--- a/src/uboot_env.c
+++ b/src/uboot_env.c
@@ -1131,7 +1131,8 @@ static int libuboot_load(struct uboot_ctx *ctx)
 	return ctx->valid ? 0 : -ENODATA;
 }
 
-#define LINE_LENGTH 1024
+/* Our certs are up to 2k, even when stripped. 10 kB should be very safe. */
+#define LINE_LENGTH 10240
 int libuboot_load_file(struct uboot_ctx *ctx, const char *filename)
 {
 	FILE *fp;
-- 
2.30.2

